src/include/stir/recon_buildblock/LogcoshPrior.h

```diff
@@ -0,0 +1,203 @@
+//
+//
+/*
+ Copyright (C) 2000- 2011, Hammersmith Imanet Ltd
+ Copyright (C) 2016, 2020, UCL
+ This file is part of STIR.
+
+ This file is free software; you can redistribute it and/or modify
+ it under the terms of the GNU Lesser General Public License as published by
+ the Free Software Foundation; either version 2.1 of the License, or
+ (at your option) any later version.
+
+ This file is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Lesser General Public License for more details.
+
+ See STIR/LICENSE.txt for details.
+ */
+/*!
+ \file
+ \ingroup priors
+ \brief Declaration of class stir::LogcoshPrior
+
+ \author Kris Thielemans
+ \author Sanida Mustafovic
+ \author Yu-Jung Tsai
+ \author Robert Twyman
+
+ */
+
+
+#ifndef __stir_recon_buildblock_LogcoshPrior_H__
+#define __stir_recon_buildblock_LogcoshPrior_H__
+
+
+#include "stir/RegisteredParsingObject.h"
+#include "stir/recon_buildblock/PriorWithParabolicSurrogate.h"
+#include "stir/Array.h"
+#include "stir/DiscretisedDensity.h"
+#include "stir/shared_ptr.h"
+#include <string>
+
+START_NAMESPACE_STIR
+
+
+/*!
+ \ingroup priors
+ \brief
+ A class in the GeneralisedPrior hierarchy. This implements a logcosh Gibbs prior.
+
+ The gradient of the prior is computed as follows:
+
+ \f[
+ g_r = \sum_dr w_{dr} tanh(\lambda_r - \lambda_{r+dr}) * \kappa_r * \kappa_{r+dr}
+ \f]
```

Comment:
I'd replace this with the logcosh formula prior itself, as opposed to giving the gradient. It will need alpha